### ISSUE TITLE
Raspian Buster

### ISSUE THREAD

**Issue by danlun** [lord2y/rtl8192eu-arm-linux-driver#4]
Im trying to follow your directions to get mt DWA-131 to work in a Pi2 with wifi dongle DWA-131 (Dlink). Its a image fron Hoobs (homebridge). When I do the DKms install i get

hoobs@hoobs:~/.hoobs/etc/rtl8192eu-linux-driver $ sudo dkms install rtl8192eu/1.0
Error! echo
Your kernel headers for kernel 4.19.75-v7+ cannot be found at
/lib/modules/4.19.75-v7+/build or /lib/modules/4.19.75-v7+/source.
hoobs@hoobs:~/.hoobs/etc/rtl8192eu-linux-driver $ 

Would you have any idea what Im supposed to do to avoid this?
Thanks

**Comment by lord2y** [comment]
Hello,

it seems you didn't install the kernel headers.
Have you tried running:

```bash
# apt-get install git raspberrypi-kernel-headers build-essential dkms
```

**Comment by danlun** [comment]
Yes I have actually but thanks for pointing it out. I did it one more time of course to double check and this came back

hoobs@hoobs:~/.hoobs/etc $ sudo apt-get install git raspberrypi-kernel-headers build-essential dkms
Reading package lists... Done
Building dependency tree       
Reading state information... Done
build-essential is already the newest version (12.6).
dkms is already the newest version (2.6.1-4).
git is already the newest version (1:2.20.1-2+deb10u1).
raspberrypi-kernel-headers is already the newest version (1.20200114-1).
0 upgraded, 0 newly installed, 0 to remove and 13 not upgraded.
hoobs@hoobs:~/.hoobs/etc $ 

//dan

**Comment by lord2y** [comment]
It seems you're running an old kernel `4.19.75-v7+` while you installed 
`raspberrypi-kernel-headers version (1.20200114-1)` which should come from
kernel version `4.19.93-v7`.
You should check whether `raspberrypi-kernel` and `raspberrypi-kernel-headers`
are the same version:
```bash
 $ dpkg-query -l | egrep -i "\-kernel|\-headers"
ii  raspberrypi-kernel             1.20200114-1                        armhf        Raspberry Pi bootloader
ii  raspberrypi-kernel-headers     1.20200114-1                        armhf        Header files for the Raspberry Pi Linux kernel
```
In case they aren't please install the latest package for `raspberrypi-kernel` and
try again.  That should hopefully solve your issue :smile:

**Comment by danlun** [comment]
Ahh, yes there was a difference which now is fixed so it looks like above. I did the run the install again, the old errors are gone but this came up.

hoobs@hoobs:~/.hoobs/etc $ sudo dkms install rtl8192eu/1.0

Kernel preparation unnecessary for this kernel.  Skipping...

Building module:
cleaning build area...
'make' all KVER=4.19.93-v7+.....(bad exit status: 2)
Error! Bad return status for module build on kernel: 4.19.93-v7+ (armv7l)
Consult /var/lib/dkms/rtl8192eu/1.0/build/make.log for more information.
hoobs@hoobs:~/.hoobs/etc $ 

I will look into the make.log and copy also.

**Comment by danlun** [comment]
This is the make.log file:

DKMS make.log for rtl8192eu-1.0 for kernel 4.19.93-v7+ (armv7l)
Sun 02 Feb 2020 05:22:30 AM MST
/bin/sh: 1: bc: not found
/bin/sh: 1: echo: echo: I/O error
make ARCH=armv7l CROSS_COMPILE= -C /lib/modules/4.19.93-v7+/build M=/var/lib/dk$
make[1]: Entering directory '/usr/src/linux-headers-4.19.93-v7+'
Makefile:614: arch/armv7l/Makefile: No such file or directory
make[1]: *** No rule to make target 'arch/armv7l/Makefile'.  Stop.
make[1]: Leaving directory '/usr/src/linux-headers-4.19.93-v7+'
make: *** [Makefile:2298: modules] Error 2

Im not a full blown techy at all, but trying my best.....
Appreciate you trying to help
//dan

**Comment by lord2y** [comment]
> This is the make.log file:
> 
> DKMS make.log for rtl8192eu-1.0 for kernel 4.19.93-v7+ (armv7l)
> Sun 02 Feb 2020 05:22:30 AM MST
> /bin/sh: 1: bc: not found
> /bin/sh: 1: echo: echo: I/O error

Looks like your system doesn't have `bc` installed.
Try 

`apt-get install bc`

and let me know

**Comment by danlun** [comment]
Correct again of course. bc now installed.

This is a HOOBS image Im running to make "homebridge" run om my PI2 and this WIFI dongle dongle, just so I dont spend all your time for no good...... This is the make.log file this time.....

DKMS make.log for rtl8192eu-1.0 for kernel 4.19.93-v7+ (armv7l)
Sun 02 Feb 2020 07:20:20 AM MST
make ARCH=armv7l CROSS_COMPILE= -C /lib/modules/4.19.93-v7+/build M=/var/lib/dk$
make[1]: Entering directory '/usr/src/linux-headers-4.19.93-v7+'
Makefile:614: arch/armv7l/Makefile: No such file or directory
make[1]: *** No rule to make target 'arch/armv7l/Makefile'.  Stop.
make[1]: Leaving directory '/usr/src/linux-headers-4.19.93-v7+'
make: *** [Makefile:2298: modules] Error 2

so some progress but still not there....

**Comment by lord2y** [comment]
Could you try running

```make arch=arm```

in the root dir of the driver?
Let's see if you have better luck :crossed_fingers:

**Comment by danlun** [comment]
OK, as it seems i have two installs (very possible because i hav etried to get this working for a while, before I found you :-)) First try gave this

hoobs@hoobs:~/.hoobs/etc/rtl8192eu-linux-driver $ make arch=arm
make ARCH=armv7l CROSS_COMPILE= -C /lib/modules/4.19.93-v7+/build M=/home/hoobs/.hoobs/etc/rtl8192eu-linux-driver  modules
make[1]: Entering directory '/usr/src/linux-headers-4.19.93-v7+'
Makefile:614: arch/armv7l/Makefile: No such file or directory
make[1]: *** No rule to make target 'arch/armv7l/Makefile'.  Stop.
make[1]: Leaving directory '/usr/src/linux-headers-4.19.93-v7+'
make: *** [Makefile:2298: modules] Error 2

Then I did the same in 

hoobs@hoobs:~/rtl8192eu-linux-driver $ make arch=arm

and off we went, building :-) so that last one worked, ending with

......
CC [M]  /home/hoobs/rtl8192eu-linux-driver/core/rtw_mp.o
  LD [M]  /home/hoobs/rtl8192eu-linux-driver/8192eu.o
  Building modules, stage 2.
  MODPOST 1 modules
  CC      /home/hoobs/rtl8192eu-linux-driver/8192eu.mod.o
  LD [M]  /home/hoobs/rtl8192eu-linux-driver/8192eu.ko
make[1]: Leaving directory '/usr/src/linux-headers-4.19.93-v7+'
hoobs@hoobs:~/rtl8192eu-linux-driver $ 

Then I try again with

hoobs@hoobs:~/.hoobs/etc $ sudo dkms install rtl8192eu/1.0

but it error again with this make.log file

DKMS make.log for rtl8192eu-1.0 for kernel 4.19.93-v7+ (armv7l)
Sun 02 Feb 2020 10:42:03 AM MST
make ARCH=armv7l CROSS_COMPILE= -C /lib/modules/4.19.93-v7+/build M=/var/lib/dk$
make[1]: Entering directory '/usr/src/linux-headers-4.19.93-v7+'
Makefile:614: arch/armv7l/Makefile: No such file or directory
make[1]: *** No rule to make target 'arch/armv7l/Makefile'.  Stop.
make[1]: Leaving directory '/usr/src/linux-headers-4.19.93-v7+'
make: *** [Makefile:2298: modules] Error 2

Am I in wrong diretory with this DKMS install command?

**Comment by lord2y** [comment]
Cool! Glad you were able to build the module.

Did you run this command in the driver's directory:

```bash
dkms add .
```

**Comment by lord2y** [comment]
I released [v4.19](https://github.com/lord2y/rtl8192eu-arm-linux-driver/releases/tag/v4.19) that should work better with kernel 4.19.xx.
You'll find attached also a deb package that should simplify the installation of the driver.
You're welcome to test it and let me know whether this will fix your issue.

**Comment by danlun** [comment]
Yes, this command was done before, but in the wrong directory (or the first one) but I tried again in the correct firectory (where make worked). But I got an error message and the dried to uninstall but didnt succeedd as you can see.

hoobs@hoobs:~/rtl8192eu-linux-driver $ sudo dkms add .
Error! DKMS tree already contains: rtl8192eu-1.0
You cannot add the same module/version combo more than once.

hoobs@hoobs:~/rtl8192eu-linux-driver $ sudo dkms uninstall -m rtl8192eu -v 1.0
Error! The module rtl8192eu 1.0 is not currently installed.
This module is not currently ACTIVE for kernel 4.19.93-v7+ (armv7l).
hoobs@hoobs:~/rtl8192eu-linux-driver $

**Comment by lord2y** [comment]
Have you tried using this deb: https://github.com/lord2y/rtl8192eu-arm-linux-driver/releases/download/v4.19/rtl8192eu-dkms_1.0_armhf.deb attached to the latest release?

**Comment by danlun** [comment]
Hi again, sorry for Bring late in reply. I have been included in other
projects for a while now. I also have ordered a standard PI WiFi dongle and
will test that before going forward.
I’m great full for all help and if I can’t get the std PI WiFi to work I
will return to this.
Thanks
Dan

tis 18 feb. 2020 kl. 11:04 skrev Alessandro Ratti <notifications@github.com

**Comment by danlun** [comment]
I went the easy way in the end. I have used up to many peoples help and unfortunately, even with all that help I got, it didnt work in the end for me. End result became I bought a PI original (white) USB Wifi dongle which the Hoobs team has as a supported one, and after restarting the PI with a new burned image it all worked out perfect as the instructions says on the Hoobs home page. So I have to use the Dlink wifi dongle somewhere else.... :-) MANY THANKS!!!!!
